src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -815,32 +806,35 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
             else
 #endif // TARGET_ARM64
             {
-                emitAttr storeAttr = emitTypeSize(targetType);
+                emitAttr storeAttr = emitTypeSize(source->TypeGet());
                 emit->emitIns_S_R(INS_str, storeAttr, srcReg, varNumOut, argOffsetOut);
                 argOffsetOut += EA_SIZE_IN_BYTES(storeAttr);
             }
             assert(argOffsetOut <= argOffsetMax); // We can't write beyond the outgoing arg area
             return;
         }
 
+        var_types slotType = genActualType(source);
```

Comment:
do I understand this change correctly that now we have such slot types:
1byte -  TYP_BYTE
2bytes - TYP_SHORT
4bytes - TYP_LONG (was TYP_INT before the change).
8bytes - TYP_LONG
?
so if we pass two `TYP_INT ` on the stack, in one 8byte stack slot, don't we override anything?

Reply:
Not totally sure I understand. `genActualType(TYP_INT)` returns `TYP_INT`, not `TYP_LONG`.

The difference before was that we used the real type of the operand to decide on how much to store on the stack. That's not right, a small-typed operand still produces a value of type `genActualType(operand)` and it is the `PUTARG_STK` node that decides how much should be put on the stack. So we had a problem for e.g. a `TYP_SHORT` indirection feeding into a `TYP_INT` parameter.

A `TYP_INT` indirection feeding into a 4 byte parameter will still only store 4 bytes on the stack.

Reply:
I see, thanks!